web/src/components/storage/ProposalPage.tsx

```diff
@@ -47,63 +238,33 @@ export default function ProposalPage() {
     if (isDeprecated) reprobe().catch(console.log);
   }, [isDeprecated, reprobe]);
 
-  const errors = useIssues("storage")
+  const systemErrors = issues
     .filter((s) => s.severity === IssueSeverity.Error)
-    .map(toValidationError);
-
-  const validProposal = errors.length === 0;
-
-  if (isDeprecated) {
-    return (
-      <Page>
-        <Page.Header>
-          <Content component="h2">{_("Storage")}</Content>
-        </Page.Header>
-
-        <Page.Content>
-          <Loading text={_("Reloading data, please wait...")} />
-        </Page.Content>
-      </Page>
-    );
-  }
+    .filter((s) => s.source === IssueSource.System);
+  const configErrors = issues
+    .filter((s) => s.severity === IssueSeverity.Error)
+    .filter((s) => s.source === IssueSource.Config);
+  const isValid = !systemErrors.length && !configErrors.length;
+  const isEditable = model && !configErrors.length;
+  const isDevicesEmpty = !devices.length;
 
   return (
     <Page>
       <Page.Header>
         <Content component="h2">{_("Storage")}</Content>
       </Page.Header>
-
       <Page.Content>
-        <Grid hasGutter>
-          <ProposalTransactionalInfo />
-          <ProposalFailedInfo />
-
-          <GridItem sm={12} xl={8}>
-            <Page.Section
-              title={_("Installation Devices")}
-              description={_(
-                "Structure of the new system, including disks to use and additional devices like LVM volume groups.",
-              )}
-              actions={
-                <>
-                  <SplitItem isFilled> </SplitItem>
-                  <SplitItem>
-                    <AddExistingDeviceMenu />
-                  </SplitItem>
-                  <SplitItem>
-                    <ConfigEditorMenu />
-                  </SplitItem>
-                </>
-              }
-            >
-              <ConfigEditor />
-            </Page.Section>
-          </GridItem>
-          <GridItem sm={12} xl={4}>
-            <EncryptionField password={""} isLoading={false} />
-          </GridItem>
-          {validProposal && <ProposalResultSection isLoading={false} />}
-        </Grid>
+        {isDeprecated && <Loading text={_("Reloading data, please wait...")} />}
+        {!isDeprecated && isDevicesEmpty && <DevicesEmptyState />}
+        {!isDeprecated && !isDevicesEmpty && !isEditable && !isValid && !model && (
+          <ConfigEmptyState />
+        )}
+        {!isDeprecated && !isDevicesEmpty && !isEditable && !isValid && model && (
+          <ErrorsEmptyState errors={configErrors} />
+        )}
+        {!isDeprecated && !isDevicesEmpty && (isEditable || isValid) && (
+          <ProposalSections isEditable={isEditable} isValid={isValid} />
+        )}
```

Comment:
I find this code organization a bit weird. The fact that `ConfigEmptyState` is not managed at the same level than `ProposalFailedInfo` is completely counterintuitive for me. Do you get what I mean?

Reply:
Sorry, I did not get you. `ProposalFailedInfo` is only shown when the layout contains the sections (config editor and/or result).